CMakeLists.txt

```diff
@@ -210,6 +210,17 @@ add_library(ypspur SHARED
   src/yprintf.c
   src/utility.c
 )
+
+add_library(ypspur-comm SHARED
+  src/communication.c
+  src/serial.c
```

Comment:
`serial.c` is better to be removed from exposed library. it has global variable managing serial connection of ypspur-coordinator.
Moving `encode_write` to `src/serial.c` would be good.